dtde/clusterlib.py

```diff
@@ -198,6 +199,17 @@ def create_cluster(
             batch_models.MetadataItem(
                 name=constants.AZB_SOFTWARE_METADATA_KEY, value=Software.spark),
         ])
+    
+    # Get user ssh key, prompt for password if necessary
+    ssh_key = ssh.get_user_public_key(ssh_key)
+    if username is not None and password is None and ssh_key is None:
+        password = getpass.getpass("Please input a password for user {0}: ".format(username))
+        confirm_password = getpass.getpass("Please confirm your password for user {0}: ".format(username))
+        if password != confirm_password:
+            raise ThunderboltError("Password confirmation did not match, please try again.")
+        if not password:
+            raise ThunderboltError(
+                "Cannot add user to cluster. Need to provide a ssh public key in .thunderbolt/secrets.yaml or password.")
```

Comment:
Can it also be done via a command line argument?

Reply:
Yeah, you can do it via command line as well. This error is actually a bit odd. This would only happen if the user didn't supply a password or ssh key for the user they are trying to create, and then when prompted to input a password, just hit enter twice (didn't input a password). So I'm not sure that this is even the best description of the issue. I'll update it to be more reflective.